### PR TITLE
Add macro theory rewrite MACRO_BOOL_NNF_NORM

### PR DESCRIPTION
**Pull request by ajreynol**
This adds a macro theory rewrite to the Boolean rewriter, which is capture NNF normalization done in the quantifiers rewriter.

Note that `MACRO_` theory rewrites will not appear in final proofs.

We move this method to the Boolean rewriter and add proof tracking for it, which will be used in an elaboration pass inside DSL reconstruction (to be added on a followup PR).

### CODE REVIEW

File: src/theory/booleans/theory_bool_rewriter.cpp [hunk 1]
```diff
@@ -32,6 +34,236 @@ TheoryBoolRewriter::TheoryBoolRewriter(NodeManager* nm) : TheoryRewriter(nm)
...
+        {
+          k = Kind::OR;
+          negAllCh = true;
+        }
+        else if (cur[0].getKind() == Kind::XOR
+                 || (cur[0].getKind() == Kind::EQUAL
+                     && cur[0][0].getType().isBoolean()))
+        {
+          k = Kind::EQUAL;
+          negCh1 = (cur[0].getKind() == Kind::EQUAL);
```
Comment (abdoo8080) [inline src/theory/booleans/theory_bool_rewriter.cpp:146]:
```suggestion
          negCh1 = cur[0].getKind() == Kind::EQUAL;
```

File: src/theory/booleans/theory_bool_rewriter.cpp [hunk 2]
```diff
@@ -32,6 +34,236 @@ TheoryBoolRewriter::TheoryBoolRewriter(NodeManager* nm) : TheoryRewriter(nm)
...
+          negCh1 = true;
+        }
+        else
+        {
+          visited[cur] = cur;
+          continue;
+        }
+        ncur = cur[0];
+      }
+      else if ((k != Kind::EQUAL || !cur[0].getType().isBoolean())
```
Comment (abdoo8080) [inline src/theory/booleans/theory_bool_rewriter.cpp:161]: `!cur[0].getType().isBoolean()` was `!body[0].getType().isBoolean()` before. Was it a bug?

File: src/proof/trust_id.h [hunk 3]
```diff
@@ -66,6 +66,10 @@ enum class TrustId : uint32_t
   QUANTIFIERS_PREPROCESS,
   /** A subtype elimination step that could not be processed */
   SUBTYPE_ELIMINATION,
+  /** A rewrite required for showing a macro theory rewrite */
+  MACRO_THEORY_REWRITE_RCONS,
```
Comment (abdoo8080) [inline src/proof/trust_id.h:70]: What is this for?

**Review by abdoo8080 — APPROVED** [review]

File: src/proof/trust_id.h (hunk 3 above)
Comment (ajreynol) [reply src/proof/trust_id.h:70]: This will be used if elaboration of a `ProofRewriteRule::MACRO_` fails.  This is not expected to ever happen.

**Pull request merged**